### PR TITLE
Adding SQL Triggers that check for undefined types in Custom XML

### PR DESCRIPTION
**Pull request by dhchandw**
Added SQL Triggers that check for undefined types in attributes ,command arguments and event fields for a custom XML when inserted into a session. Add session notice as required

Added relevant tests

Corrected typo in package.json

### CODE REVIEW

**Review by brdandu — APPROVED** [review]

File: test/custom-matter-xml.test.js [hunk 1]
```diff
@@ -449,6 +449,26 @@ test(
...
+          notif.message.includes('Undefined Data Type') &&
+          notif.message.includes('Attribute Sample Mfg Specific Attribute') &&
+          notif.message.includes('matter-bad-custom.xml')
+      )
+    ).toBeTruthy()
+    expect(
+      sessionNotif.some(
+        (notif) =>
+          notif.type === 'ERROR' &&
+          notif.message.includes('Undefined Data Type') &&
```
[the remark this answers is not in the document]
Comment (dhchandw) [reply test/custom-matter-xml.test.js:467]: intentional errors in my custom XML with bad data type

File: test/custom-matter-xml.test.js [hunk 2]
```diff
@@ -449,6 +449,26 @@ test(
     expect(
       packageNotif.some((notif) => notif.message.includes('type contradiction'))
     ).toBeTruthy() // checks if the correct warning is thrown
+
+    let sessionNotif = await querySessionNotification.getNotification(db, sid)
+    expect(
+      sessionNotif.some(
+        (notif) =>
+          notif.type === 'ERROR' &&
+          notif.message.includes('Undefined Data Type') &&
```
[the remark this answers is not in the document]
Comment (dhchandw) [reply test/custom-matter-xml.test.js:458]: intentional errors in my custom XML with bad data type

File: src-electron/db/zap-schema.sql [hunk 3]
```diff
@@ -3925,15 +4421,10 @@ BEGIN
     )
   AND
     NOTICE_TYPE = 'ERROR'
-  AND
-    NOTICE_MESSAGE LIKE '%code conflict%'
```
[the remark this answers is not in the document]
Comment (dhchandw) [reply src-electron/db/zap-schema.sql:3929]: This was specifically deleting for certain notification.  However, now it's being used to delete any session specific custom xml notifs when the relevant session package is disabled/deleted.

File: src-electron/db/zap-schema.sql [hunk 4]
```diff
@@ -3902,14 +4400,12 @@ BEGIN
     )
   AND
     NOTICE_TYPE = "ERROR"
-  AND
-    NOTICE_MESSAGE LIKE '%code conflict%'
```
[the remark this answers is not in the document]
Comment (dhchandw) [reply src-electron/db/zap-schema.sql:3906]: This was specifically deleting for certain notification.  However, now it's being used to delete any session specific custom xml notifs when the relevant session package is disabled/deleted.

**Comment by ethanzhouyc** [comment]
What does the second new commit do? I did not see it change anything

**Review by ethanzhouyc — APPROVED** [review]

**Pull request merged**